### PR TITLE
Fix invalid buffered column to column id mapping

### PR DESCRIPTION
**Pull request by MBkkt**

### CODE REVIEW

**Review by Dronplane — COMMENTED** [review]
Is it possible to add a test that reproduces the issue?

File: core/index/segment_writer.hpp [hunk 1]
```diff
@@ -164,21 +165,19 @@ class segment_writer : public ColumnProvider, util::noncopyable {
...
-
     return nullptr;
   }
 
   const column_reader* column(std::string_view name) const final {
     const auto it = columns_.find(hashed_string_view{name});
-
-    if (it == columns_.end()) {
-      return nullptr;
+    if (it != columns_.end()) {
```
[the remark this answers is not in the document]
Comment (MBkkt) [reply core/index/segment_writer.hpp:177]: Yep

**Comment by MBkkt** [comment]
> Is it possible to add a test that reproduces the issue?

I'm sure it's possible, but now we have tests in arangodb which cover this.
So plan is fix tests in arangodb, and then quietly add tests in iresearch.
The issue we need new wand data in 3.11 format as soon as possible
Also this issue block adding new wand test by @alexbakharew

File: CMakeLists.txt [hunk 2]
```diff
@@ -13,6 +13,7 @@ endif ()
 
 set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_CXX_EXTENSIONS OFF)
```
Comment (MBkkt) [inline CMakeLists.txt:16]: https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html#prop_tgt:CXX_EXTENSIONS

**Pull request merged**